### PR TITLE
Changes web scraper to implement a new, more effective PDF scraping algorithm

### PR DESCRIPTION
**Pull request by SoftDrink**
- Use PDFMiner instead of Textract
- Switch to Python 3
- Implements tests (and contracts for spiders)
- Detects sections by layout analysis instead of regex
- More easily reusable PDF objects
- Settings tuning for better performances and to avoid bottlenecks
- DSX feed storage to send scrapes directly into IBM object storage
- Incremental queue using sqlite

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,52 +11,18 @@ RUN apt-get update -y
...
-    python-pyrex
+    gcc \
+    python3-pip \
 
 # Update pip
-RUN pip install --upgrade pip
+RUN pip3 install --upgrade pip
 # Install any needed packages specified in requirements.txt
-RUN pip install scrapy textract pybloom botocore
+RUN pip3 install scrapy pdfminer.six botocore pybloom_live requests
```
Comment (jakubfiala) [inline Dockerfile]: I feel like with 5+ dependencies installed with `pip` you might wanna make a `requirements.txt`

File: pdf_parser/tools/extraction.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        max_fonts_name = line.font_face
+        break
+
+    mean_fonts = pdf_file.get_mean_font_size()
+    upper_mean = pdf_file.get_upper_mean_font_size()
+    for fsize in list_fonts:
+        for line in pdf_file.get_lines_by_font_size(fsize):
+            cnd_a = fsize >= upper_mean + (upper_mean - mean_fonts)
+            cnd_b = cnd_a and line.bold
+            cnd_c = fsize > upper_mean + 2 and line.font_face == max_fonts_name
```
Comment (jakubfiala) [inline pdf_parser/tools/extraction.py]: these variable names could be more descriptive, what does `cnd` stand for?

Comment (SoftDrink) [reply pdf_parser/tools/extraction.py]: They could definitely. The file also needs some comments. I'll change that

**Comment by nsorros** [comment]
Wow, that is a big PR. It would help to break it down a bit in the future.

**Comment by SoftDrink** [comment]
Yep I know, it's the first major change to the web scrapper, and it implements a lot of new things... Sorry

**Pull request merged**